google/resource_google_project_iam_custom_role.go

```diff
@@ -124,12 +138,23 @@ func resourceGoogleProjectIamCustomRoleUpdate(d *schema.ResourceData, meta inter
 			if err := resourceGoogleProjectIamCustomRoleDelete(d, meta); err != nil {
 				return err
 			}
-		} else {
+			d.SetPartial("deleted")
+		}
+	}
+
+	// If role is not deleted, make sure it exists and undelete if needed.
```

Comment:
Same comment Dana pointed out about the unnecessary PATCH applies here, I think.